File: VB/CustomRecurrenceDialogViewModel.vb
Imports DevExpress.Xpf.Scheduler.UI
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraScheduler

Namespace MVVMSchedulerApplication
    Public Class CustomRecurrenceDialogViewModel
        Inherits RecurrenceDialogViewModel

        Public Sub New(ByVal viewModel As AppointmentFormViewModel)
            MyBase.New(viewModel)

        End Sub
        Private Sub HideRecurrenceTypes()
            Dim availableRecurrenceTypes As System.Collections.ObjectModel.ObservableCollection(Of RecurrenceType) = Me.AvailableRecurrenceTypes
            For i As Integer = availableRecurrenceTypes.Count - 1 To 0 Step -1
                If availableRecurrenceTypes(i) = DevExpress.XtraScheduler.RecurrenceType.Yearly OrElse availableRecurrenceTypes(i) = DevExpress.XtraScheduler.RecurrenceType.Monthly Then
                    availableRecurrenceTypes.RemoveAt(i)
                End If
            Next i
        End Sub
        Private Sub SetPredefinedValues()
            Me.RecurrenceInfo.WeekDays = WeekDays.WorkDays
            Me.RecurrenceInfo.Range = RecurrenceRange.OccurrenceCount
            Me.RecurrenceInfo.OccurrenceCount = 5
        End Sub
        Public Sub New(ByVal viewModel As AppointmentFormViewModel, ByVal [readOnly] As Boolean)
            MyBase.New(viewModel, [readOnly])
            If viewModel.IsNewAppointment Then
                SetPredefinedValues()
                HideRecurrenceTypes()

            End If
        End Sub
        Public Sub New(ByVal viewModel As AppointmentFormViewModel, ByVal availableRecurrenceTypes() As DevExpress.XtraScheduler.RecurrenceType)
            MyBase.New(viewModel, availableRecurrenceTypes)

        End Sub
        Public Sub New(ByVal viewModel As AppointmentFormViewModel, ByVal availableRecurrenceTypes() As DevExpress.XtraScheduler.RecurrenceType, ByVal [readOnly] As Boolean)
            MyBase.New(viewModel, availableRecurrenceTypes, [readOnly])

        End Sub

    End Class
End Namespace

File: CS/CustomRecurrenceDialogViewModel.cs
using DevExpress.Xpf.Scheduler.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraScheduler;

namespace MVVMSchedulerApplication
{
    public class CustomRecurrenceDialogViewModel : RecurrenceDialogViewModel
    {
        public CustomRecurrenceDialogViewModel(AppointmentFormViewModel viewModel)
            : base(viewModel)
        {

        }
        private void HideRecurrenceTypes()
        {
            System.Collections.ObjectModel.ObservableCollection<RecurrenceType> availableRecurrenceTypes = this.AvailableRecurrenceTypes;
            for (int i = availableRecurrenceTypes.Count - 1; i >= 0; i--)
            {
                if (availableRecurrenceTypes[i] == DevExpress.XtraScheduler.RecurrenceType.Yearly ||
                    availableRecurrenceTypes[i] == DevExpress.XtraScheduler.RecurrenceType.Monthly)
                    availableRecurrenceTypes.RemoveAt(i);
            }
        }
        private void SetPredefinedValues()
        {
            this.RecurrenceInfo.WeekDays = WeekDays.WorkDays;
            this.RecurrenceInfo.Range = RecurrenceRange.OccurrenceCount;
            this.RecurrenceInfo.OccurrenceCount = 5;
        }
        public CustomRecurrenceDialogViewModel(AppointmentFormViewModel viewModel, bool readOnly)
            : base(viewModel, readOnly)
        {
            if (viewModel.IsNewAppointment)
            {
                SetPredefinedValues();
                HideRecurrenceTypes();

            }
        }
        public CustomRecurrenceDialogViewModel(AppointmentFormViewModel viewModel, DevExpress.XtraScheduler.RecurrenceType[] availableRecurrenceTypes)
            : base(viewModel, availableRecurrenceTypes)
        {

        }
        public CustomRecurrenceDialogViewModel(AppointmentFormViewModel viewModel, DevExpress.XtraScheduler.RecurrenceType[] availableRecurrenceTypes, bool readOnly)
            : base(viewModel, availableRecurrenceTypes, readOnly)
        {

        }

    }
}
